.github/workflows/goreleaser.yml

```
@@ -7,12 +7,15 @@ on:
 
 jobs:
   goreleaser-check:
+    if: github.event_name == 'release'
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
       - run: git fetch --force --tags
+      - name: Set GORELEASER_CURRENT_TAG in GitHub env
+        run: echo "GORELEASER_CURRENT_TAG=${{ github.event.release.tag_name }}" >> $GITHUB_ENV
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC2086:info:1:71: Double quote to prevent globbing and word splitting

(shellcheck)